pkg/render/logstorage/esmetrics/elasticsearch_metrics.go

```diff
@@ -209,6 +211,39 @@ func (e elasticsearchMetrics) metricsDeployment() *appsv1.Deployment {
 		annotations[e.cfg.ServerTLS.HashAnnotationKey()] = e.cfg.ServerTLS.HashAnnotationValue()
 	}
 
+	_, esHost, esPort, _ := url.ParseEndpoint(relasticsearch.GatewayEndpoint(e.SupportedOSType(), e.cfg.ClusterDomain, render.ElasticsearchNamespace))
+
+	envVars := []corev1.EnvVar{
+		relasticsearch.ElasticUsernameEnvVar(ElasticsearchMetricsSecret),
+		relasticsearch.ElasticPasswordEnvVar(ElasticsearchMetricsSecret),
+		relasticsearch.ElasticHostEnvVar(esHost),
+		relasticsearch.ElasticPortEnvVar(esPort),
+		relasticsearch.ElasticCAEnvVar(e.SupportedOSType()),
+	}
+
+	container := corev1.Container{
+		Name:            ElasticsearchMetricsName,
+		Image:           e.esMetricsImage,
+		ImagePullPolicy: render.ImagePullPolicy(),
+		SecurityContext: securitycontext.NewNonRootContext(),
+		Command:         []string{"/bin/elasticsearch_exporter"},
+		Args: []string{
+			"--es.uri=https://$(ELASTIC_USERNAME):$(ELASTIC_PASSWORD)@$(ELASTIC_HOST):$(ELASTIC_PORT)",
+			"--es.all", "--es.indices", "--es.indices_settings", "--es.shards", "--es.cluster_settings",
+			"--es.timeout=30s", "--es.ca=$(ELASTIC_CA)", "--web.listen-address=:9081",
+			"--web.telemetry-path=/metrics", "--tls.key=/tigera-ee-elasticsearch-metrics-tls/tls.key", "--tls.crt=/tigera-ee-elasticsearch-metrics-tls/tls.crt", fmt.Sprintf("--ca.crt=%s", certificatemanagement.TrustedCertBundleMountPath),
+		},
+		VolumeMounts: append(
+			e.cfg.TrustedBundle.VolumeMounts(e.SupportedOSType()),
+			e.cfg.ServerTLS.VolumeMount(e.SupportedOSType()),
+		),
+		Env: []corev1.EnvVar{
+			{Name: "FIPS_MODE_ENABLED", Value: operatorv1.IsFIPSModeEnabledString(e.cfg.Installation.FIPSMode)},
+		},
+	}
+
+	container.Env = append(container.Env, envVars...)
```

Comment:
I think throughout we should clean up this split env declaration - it's a bit confusing the way this is laid out here.

- Define envVars
- Create a container with a separate set of hard-coded env vars
- Append the previously created env vars to the container's env.

We should just be able to define all of the env vars in one place.

Reply:
Agreed that's what I have in my current iteration.